bin/osg-notify

```diff
@@ -144,6 +147,22 @@ def network_ok(bypass_dns_check):
         print("  https://opensciencegrid.org//operations/services/sending-announcements/")
         return False
 
+def replace_smart_quotes_and_dashes(contents):
+    # Replace smart quotes and em/en dashes
+    replaced = contents.replace('“','"').replace('”','"').replace("–","-").replace("—","-")
+    return replaced
+
+def has_non_printable_ascii_characters(contents):
```

Comment:
Diego i'm sorry i prompted you to mess with this so much.  It should work fine as you have it, especially since the main use-case it to catch (non-ascii) unicode codepoints, which are all > 126.

Though i'm just gonna throw it out there, this whole function could perhaps be equivalently written as

```
def has_non_ascii(s):
    return re.search(r'[^ -~\t\n]', s) is not None
```

that is, if the search for a regex pattern {that contains any characters _not_ in the character set of the printable ascii-range and tab and newline} matches, return true, else false.  I think it's identical to what you have, just short-hand with the regex.  Feel free to use whichever you like best.

Reply:
Thanks fo the suggestion Carl but if both functions get the same result, I will just not bother to make another change ;)